src/Symfony/Bundle/WebServerBundle/WebServerConfig.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\WebServerBundle;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class WebServerConfig
+{
+    private $hostname;
+    private $port;
+    private $documentRoot;
+    private $env;
+    private $router;
+
+    public function __construct($documentRoot, $env, $address = null, $router = null)
+    {
+        if (!is_dir($documentRoot)) {
+            throw new \InvalidArgumentException(sprintf('The document root directory "%s" does not exist.', $documentRoot));
+        }
+
+        if (null === $file = $this->guessFrontController($documentRoot, $env)) {
+            throw new \InvalidArgumentException(sprintf('Unable to guess the front controller under "%s".', $documentRoot));
+        }
+
+        putenv('APP_FRONT_CONTROLLER='.$file);
+
+        $this->documentRoot = $documentRoot;
+        $this->env = $env;
+        $this->router = $router ?: __DIR__.'/Resources/router.php';
+
+        if (null === $address) {
+            $this->address = '127.0.0.1';
+            $this->port = $this->findBestPort();
+            $address = $this->address.':'.$this->port;
+        }
+
+        if (false !== $pos = strrpos($address, ':')) {
+            $this->hostname = substr($address, 0, $pos);
+            $this->port = substr($address, $pos + 1);
+        } elseif (ctype_digit($address)) {
```

Comment:
Should we check if the user is trying to use a privileged port (i.e. < 1024) ?

Otherwise, you'll end up with this error:

![command-port-1](https://cloud.githubusercontent.com/assets/73419/21694314/e0dbdf50-d384-11e6-8305-428361797563.png)

![browser-port-1](https://cloud.githubusercontent.com/assets/73419/21694318/e37ed35c-d384-11e6-8b81-e498160a97c0.png)